packages/web-components/fast-foundation/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,469 @@
+                this.horizontalInset = "true";
+                this.verticalInset = "false";
+                this.horizontalScaling = "anchor";
+                this.verticalScaling = "content";
+                break;
+        }
+    }
+
+    /**
+     *  Gets the anchor element by id
+     */
+    private getAnchor = (): HTMLElement | null => {
+        return document.getElementById(this.anchor);
+    };
+
+    /**
+     * handles key down events to check for dismiss
+     */
+    private handleDocumentKeydown = (e: KeyboardEvent): void => {
+        if (!e.defaultPrevented && this.tooltipVisible) {
+            switch (e.keyCode) {
+                case keyCodeEscape:
+                    this.isAnchorHovered = false;
+                    this.updateTooltipVisibility();
+                    this.$emit("dismiss");
+                    break;
+            }
+        }
+    };
+
+    /**
+     * determines whether to show or hide the tooltip based on current state
+     */
+    private updateTooltipVisibility = (): void => {
+        if (this.visible === false) {
+            this.hideTooltip();
+        } else if (this.visible === true) {
+            this.showTooltip();
+        } else {
+            if (this.isAnchorHovered) {
+                this.showTooltip();
+                return;
+            }
+            this.hideTooltip();
+        }
+    };
+
+    /**
+     * shows the tooltip
+     */
+    private showTooltip = (): void => {
+        if (this.tooltipVisible) {
+            return;
+        }
+        this.currentDirection = this.getDirection();
+        this.tooltipVisible = true;
+        document.addEventListener("keydown", this.handleDocumentKeydown);
+        DOM.queueUpdate(this.setRegionProps);
+    };
+
+    /**
+     * hides the tooltip
+     */
+    private hideTooltip = (): void => {
+        if (!this.tooltipVisible) {
+            return;
+        }
+        if (this.region !== null && this.region !== undefined) {
+            (this.region as any).removeEventListener("change", this.handlePositionChange);
+            this.region.viewportElement = null;
+            this.region.anchorElement = null;
+        }
+        document.removeEventListener("keydown", this.handleDocumentKeydown);
+        this.tooltipVisible = false;
+    };
+
+    /**
+     * updates the tooltip anchored region props after it has been
+     * added to the DOM
+     */
+    private setRegionProps = (): void => {
+        if (!this.tooltipVisible) {
+            return;
+        }
+        this.viewportElement = document.body;
+        this.region.viewportElement = this.viewportElement;
+        this.region.anchorElement = this.anchorElement;
+        (this.region as any).addEventListener("change", this.handlePositionChange);
+    };
+
+    /**
+     *  gets the current direction
+     */
+    private getDirection = (): Direction => {
+        const closest: Element | null = this.closest(
```

Comment:
There is a utility in fast-foundation for this now; `getDirection`

Reply:
switched to that, thx!